robottelo/hosts.py

```diff
@@ -52,6 +53,23 @@ class ContentHostError(Exception):
 
 class ContentHost(Host):
     run = Host.execute
+    subscribed = False
+
+    @cached_property
+    def nailgun_host(self):
+        """If this host is subscribed, provide access the it's nailgun object"""
+        if self.subscribed:
+            return entities.Host().search(query={'search': self.hostname})[0]
+
+    @property
+    def subscribed(self):
+        """Boolean representation of a content host's subscription status"""
+        return 'Status: Unknown' not in self.execute('subscription-manager status').stdout
```

Comment:
Any specific reason for adding indirect approach that host is registered, you can also able to verify the status as `Status: Current` or `subscription-manager list | egrep -i 'status: '`

Reply:
There are multiple statuses possible, 

- fully subscribed: `Overall Status: Current` (return code 0)
- partially or fully unsubscribed: `Overall Status: Invalid` (return code 1)
- unregistered: `Overall Status: Unknown` (return code 1)

Reply:
Tasos nailed it. The second status is very common since we also have tests where we only have custom subscriptions.